tests/dspy/test_save.py

```diff
@@ -0,0 +1,197 @@
+import json
+
+import dspy
+import dspy.teleprompt
+
+import mlflow
+from mlflow.models import ModelSignature
+from mlflow.types.schema import ColSpec, Schema
+
+from tests.helper_functions import expect_status_code, pyfunc_serve_and_score_model
+
+
+def test_basic_save():
+    class CoT(dspy.Module):
+        def __init__(self):
+            super().__init__()
+            self.prog = dspy.ChainOfThought("question -> answer")
+
+        def forward(self, question):
+            return self.prog(question=question)
+
+    dspy_model = CoT()
+    dspy.settings.configure(lm=dspy.OpenAI(model="gpt-4o-mini", max_tokens=250))
+
+    with mlflow.start_run() as run:
+        mlflow.dspy.log_model(dspy_model, "model")
+
+    # Clear the lm setting to test the loading logic.
+    dspy.settings.configure(lm=None)
```

Comment:
Can we create a fixture that, upon test exit, resets the state of the singleton?

Reply:
good call, done!